src/braket/default_simulator/simulator.py

```diff
@@ -386,6 +388,39 @@ def _observable_hash(observable: Observable) -> Union[str, dict[int, str]]:
         else:
             return str(observable.__class__.__name__)
 
+    def _map_circuit_to_contiguous_qubits(self, circuit: Circuit) -> Circuit:
+        """
+        Maps the qubits in operations and result types to contiguous qubits.
+
+        Args:
+            circuit (Circuit): The circuit containing the operations and result types.
+
+        Returns:
+            Circuit: The circuit with qubits in operations and result types mapped 
+            to contiguous qubits.
+        """
+        qubit_map = self._contiguous_qubit_mapping(circuit.qubit_set)
+
+        new_instructions = []
+        for ins in circuit.instructions:
+            new_ins = deepcopy(ins)
+            new_ins._targets = tuple([qubit_map[q] for q in ins.targets])
+            new_instructions.append(new_ins)
+
+        new_results = []
+        for result in circuit.results:
+            new_result = deepcopy(result)
```

Comment:
Instead of `deepcopy`ing the instruction, how about constructing the mapping as the IR is translated? This prevents having to create a new circuit altogether.